### ISSUE TITLE
command line: cpptraj -p ../a.prmtop -y A_struc_* -x test.nc

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#39]
I got this error
`Unrecognized input on command line: 5: A_struc_10.mdcrd`

Does `cpptraj` support load a bunch of files? if not, do you think it's good idea to add to it?

**Comment by drroe** [comment]
It does; I think the issue you are hitting is that the shell command interpreter is expanding your wildcards. Try protecting your wildcard argument with single quotes, e.g.
```
cpptraj -p ../a.prmtop -y 'A_struc_*' -x test.nc
```

**Comment by hainm** [comment]
okie.

**Comment by hainm** [comment]
@drroe feel free to reopen it if cpptraj man has not mentioned the quote.

**Comment by swails** [comment]
This is really a shell misunderstanding, not a cpptraj issue.  The cpptraj help is very clear about what it accepts:

```
[swails@Robin ~/src/ParmEd (decomp) ]$ cpptraj --help

Usage: cpptraj [-p <Top0>] [-i <Input0>] [-y <trajin>] [-x <trajout>]
               [-c <reference>]
               [-h | --help] [-V | --version] [--defines] [-debug <#>]
               [--interactive] [--log <logfile>] [-tl]
               [-ms <mask>] [-mr <mask>] [--mask <mask>] [--resmask <mask>]
       cpptraj <Top> <Input>
	-p <Top0>        : Load <Top0> as a topology file. May be specified more than once.
	-i <Input0>      : Read input from <Input0>. May be specified more than once.
	-y <trajin>      : Read from trajectory file <trajin>; same as input 'trajin <trajin>'.
	-x <trajout>     : Write trajectory file <trajout>; same as input 'trajout <trajout>'.
	-c <reference>   : Read <reference> as reference coordinates; same as input 'reference <reference>'.
	-h | --help      : Print command line help and exit.
	-V | --version   : Print version and exit.
	--defines        : Print compiler defines and exit.
	-debug <#>       : Set global debug level to <#>; same as input 'debug <#>'.
	--interactive    : Force interactive mode.
	--log <logfile>  : Record commands to <logfile> (interactive mode only). Default is 'cpptraj.log'.
	-tl              : Print length of trajectories specified with '-y' to STDOUT.
	-ms <mask>       : Print selected atom numbers to STDOUT.
	-mr <mask>       : Print selected residue numbers to STDOUT.
	--mask <mask>    : Print detailed atom selection to STDOUT.
	--resmask <mask> : Print detailed residue selection to STDOUT.
```

The shell expands the wildcards before it's ever fed to the program.  You can verify this via the command "set -x", which in bash will expand *everything* the shell executes.  See my example:

```
swails@batman /iscratch/swails/06.12.2014_CYS_Tests $ cpptraj -p cys.parm7 -y restrt.c*
+ cpptraj -p cys.parm7 -y restrt.cphmd restrt.cym restrt.cymparm_cymtraj restrt.cys restrt.cysparm_cymtraj
  Unrecognized input on command line: 5: restrt.cym

Usage: cpptraj [-p <Top0>] [-i <Input0>] [-y <trajin>] [-x <trajout>]
               [-c <reference>]
               [-h | --help] [-V | --version] [--defines] [-debug <#>]
[snip irrelevant]
```

So you can see the command that actually gets executed:

```
+ cpptraj -p cys.parm7 -y restrt.cphmd restrt.cym restrt.cymparm_cymtraj restrt.cys 
```

In this case, canceling a wild-card on the command-line so that the *program* can glob it is a little bit of an anti-pattern.  I get the feeling that it works primarily because Dan wanted `trajin` to perform globbing as though you were in the command-line (which is one of the first feature requests I ever made of Dan :)), and the `-y <arg>` uses the same functionality.  Not because he wanted CL convenience.

I'd suggest that the appropriate thing to do regarding wildcards is either:

1. Change the command-line parsing so that un-claimed arguments are processed as input trajectories so that the *shell* globbing works as expected
2. Don't document *any* use of CL wildcards, since the way cpptraj handles them is unique in my experience (and therefore confusing).  Instead, document what the shell does and why it doesn't work as some people would expect, but leave out the quirky workaround that accidentally gets it to work.

Just my 2c.

**Comment by hainm** [comment]
> Change the command-line parsing so that un-claimed arguments are processed as input trajectories so that the shell globbing works as expected

add 1c from me.

**Comment by swails** [comment]
@hainm -- that gets my vote as well.  Another alternative is to have `-y` accept an arbitrary number of arguments and simply stop when it hits an arg that starts with '-'.  I think that's what `argparse` does for flags that take an unknown number of arguments.

**Comment by wiseovvl** [comment]
can we please allow the sbatch commands to have the same start/stop/stride options as the regular trajectory file commands do?

**Comment by drroe** [comment]
> can we please allow the sbatch commands to have the same start/stop/offset options as the regular trajectory file commands do?

@wiseovvl Do you mean adding command line flags like e.g. 
```
cpptraj -p topo -y traj --trajargs 1,34,2
```
where start=1, stop=34, and 2=offset?

**Comment by wiseovvl** [comment]
yes, that would make things a lot simpler

**Comment by drroe** [comment]
Addressed by #614
